File: src/TestJava/MissingInteger.java
package TestJava;

public class MissingInteger {
        public static void main(String[] args) {
            // Example array with one missing integer
            int[] array = {1, 2, 3, 5, 6, 7, 8,9, 10};
            int missingNumber = MissingNumber(array);
            System.out.println("Missing number: " + missingNumber);
        }

        public static int MissingNumber(int[] array) {
            int n = array.length+1;
            int expectedSum = n * (n + 1) / 2;
            int actualSum = 0;
            for (int num : array) {
                actualSum += num;
            }
            return expectedSum - actualSum;
        }
    }

File: src/KotlinSamples/MissingInteger.kt
package KotlinSamples


    fun main(args: Array<String>) {
        // Example array with one missing integer
        val array = intArrayOf(1, 2, 3, 5, 6, 7, 8, 9, 10)
        val missingNumber = MissingNumber(array)
        println("Missing number: $missingNumber")
    }

    fun MissingNumber(array: IntArray): Int {
        val n = array.size + 1
        val expectedSum = n * (n + 1) / 2
        var actualSum = 0
        for (num in array) {
            actualSum += num
        }
        return expectedSum - actualSum
    }

